upup/pkg/fi/cloudup/awstasks/launchtemplate_target_terraform.go

```diff
@@ -262,55 +265,42 @@ func (t *LaunchTemplate) RenderTerraform(target *terraform.TerraformTarget, a, e
 			}
 		}
 	}
+
 	devices, err := e.buildRootDevice(cloud)
 	if err != nil {
 		return err
 	}
-	for n, x := range devices {
-		tf.BlockDeviceMappings = append(tf.BlockDeviceMappings, &terraformLaunchTemplateBlockDevice{
-			DeviceName: fi.PtrTo(n),
-			EBS: []*terraformLaunchTemplateBlockDeviceEBS{
-				{
-					DeleteOnTermination: fi.PtrTo(true),
-					Encrypted:           x.EbsEncrypted,
-					KmsKeyID:            x.EbsKmsKey,
-					IOPS:                x.EbsVolumeIops,
-					Throughput:          x.EbsVolumeThroughput,
-					VolumeSize:          x.EbsVolumeSize,
-					VolumeType:          fi.PtrTo(string(x.EbsVolumeType)),
-				},
-			},
-		})
+
+	devicesKeys := getSortedKeysFromMap(devices)
+	for _, key := range devicesKeys {
+		v := devices[key]
+		terraformLaunchTemplateBlockDevice := createTerraformLaunchTemplateBlockDevice(key, v)
+		tf.BlockDeviceMappings = append(tf.BlockDeviceMappings, terraformLaunchTemplateBlockDevice)
 	}
+
 	additionals, err := buildAdditionalDevices(e.BlockDeviceMappings)
 	if err != nil {
 		return err
 	}
-	for n, x := range additionals {
-		tf.BlockDeviceMappings = append(tf.BlockDeviceMappings, &terraformLaunchTemplateBlockDevice{
-			DeviceName: fi.PtrTo(n),
-			EBS: []*terraformLaunchTemplateBlockDeviceEBS{
-				{
-					DeleteOnTermination: fi.PtrTo(true),
-					Encrypted:           x.EbsEncrypted,
-					IOPS:                x.EbsVolumeIops,
-					Throughput:          x.EbsVolumeThroughput,
-					KmsKeyID:            x.EbsKmsKey,
-					VolumeSize:          x.EbsVolumeSize,
-					VolumeType:          fi.PtrTo(string(x.EbsVolumeType)),
-				},
-			},
-		})
+
+	additionalsKeys := getSortedKeysFromMap(additionals)
+	for _, key := range additionalsKeys {
+		v := additionals[key]
+		terraformLaunchTemplateBlockDevice := createTerraformLaunchTemplateBlockDevice(key, v)
+		tf.BlockDeviceMappings = append(tf.BlockDeviceMappings, terraformLaunchTemplateBlockDevice)
 	}
 
 	devices, err = buildEphemeralDevices(cloud, fi.ValueOf(e.InstanceType))
 	if err != nil {
 		return err
 	}
-	for n, x := range devices {
+
+	devicesKeys = getSortedKeysFromMap(devices)
+	for _, key := range devicesKeys {
+		v := devices[key]
```

Comment:
This should not be needed starting Go 1.22:
https://go.dev/blog/loopvar-preview

Reply:
Done! (inlined v)